### PR TITLE
NAT64 support

### PR DESCRIPTION
**Pull request by eqvinox**
This implements #721.  ~It's not really polished but does work.~

~TODO:~

- ~clean up code formatting~
- ~make NAT64 prefix configurable instead of hardcoding it~
- ~tests~

All done, maybe? :laughing:

### CODE REVIEW

**Comment by eqvinox** [comment]
Sidenote: I'm perfectly happy with anyone rewriting, mangling, whatever else, this code in order to get it merged. I'll try to respond to comments/change requests/… but if it's something you can just fix in a few minutes, please, go ahead!

File: iterator/iterator.c [hunk 1]
```diff
@@ -2811,6 +2812,32 @@ processQueryTargets(struct module_qstate* qstate, struct iter_qstate* iq,
...
+	if (ie->use_nat64 && real_addr.ss_family == AF_INET) {
+		struct sockaddr_in *sin = (struct sockaddr_in *)&target->addr;
+		struct sockaddr_in6 *sin6;
+
+		real_addr = ie->nat64_prefix_addr;
+		real_addrlen = ie->nat64_prefix_addrlen;
+
+		/* - NAT64 prefix length really doesn't matter at all,
+		 *   it's just a bunch of zero padding
+		 * - config validation requires the NAT64 prefix to be /96
```
Comment (eqvinox) [inline iterator/iterator.c]: Whoops, I completely forgot that RFC6052 actually specifies a different encoding for the prefix lengths. Fixing that in a minute.

**Comment by eqvinox** [comment]
Fixed non-/96 prefixes & added a test using a /48.

**Comment by momoka0122y** [comment]
When this feature is used prefer-ip6 shoud be changed to  "prefer-ip6: yes" instead of the default.
Or else it may synthesize and IPv4 address even when there is an IPv6 address.

**Comment by gthess** [comment]
@momoka0122y in principle you are right and I was thinking of implicitly enabling the `prefer-ip6` option but on second thought I would leave it to the user to explicitly use that.
Unbound keeps track of DNS RTT values to the different upstream IPs and it may be the case that a synthesized IPv4 connection is more favorable than an IPv6 one.
I will add your remark in the documentation though.

**Comment by agowa** [comment]
That sounds suspiciously like RFC 8305 behavior (Sections 3 and 4). https://www.rfc-editor.org/rfc/rfc8305#section-3

And section 7 would explain how to handle NAT64/DNS64 in that context.

**Review by gthess — APPROVED** [review]
LGTM!
Some minor fixes, formatting and refactoring that I will address on a follow up commit.
Thanks for this!

**Pull request merged**